pynicotine/shares.py

```diff
@@ -705,10 +702,15 @@ def send_num_shared_folders_files(self):
                 sharedfolders = len(list(shared))
                 sharedfiles = len(list(index))
 
-            self.queue.append(slskmessages.SharedFoldersFiles(sharedfolders, sharedfiles))
+            if self.core and self.core.logged_in:
+                self.queue.append(slskmessages.SharedFoldersFiles(sharedfolders, sharedfiles))
```

Comment:
Perhaps it would be better to split this function into count_num_shared_folders_files() and send_num_shared_folders_files()